src/main/java/com/internet/shop/controller/RegistrationController.java

```diff
@@ -36,7 +37,9 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
         String passwordRep = req.getParameter("pswRep");
 
         if (password.equals(passwordRep)) {
-            User user = new User(name, login, password, Set.of(Role.of("USER")));
+            byte[] salt = HashUtil.getSalt();
+            String hashPassword = HashUtil.hashPassword(password, salt);
+            User user = new User(name, login, hashPassword, salt, Set.of(Role.of("USER")));
```

Comment:
Maybe better to move this logic into create method in UserServiceImpl class

Reply:
done